### PR TITLE
Adds Physically Obstructive quirk

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Adds the Physically Obstructive negative quirk. A person with this
quirk can't swap places with other people when moving, akin to someone
always being in non-help intent.
/:cl:

Adds the TRAIT_NOMOBSWAP to accomplish this.

### CODE REVIEW

**Comment by MMMiracles** [comment]
how to die during blob 101

**Comment by AdamElTablawy** [comment]
this is probably the griefiest trait known to man

**Comment by 4dplanner** [comment]
You should probably let them swap if they're restrained

**Comment by 4dplanner** [comment]
I think this should come after the check of being restrained, as being freed every time an assistant crashes into you is quite powerful for a negative trait.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm big boned, okay??

**Comment by BeeSting12** [comment]
This makes people valid right?

**Comment by Ratvar** [comment]
People will get this trait and crawl if they really need to swap_places_with_people_on_help_intent.
Free points. Neutral trait, maybe?

**Comment by Farquaar** [comment]
This is probably not a good idea.

**Comment by kriskog** [comment]
Why is this a negative quirk? Free point!

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
See Mickyan Review

**Comment by ExcessiveUseOfCobblestone** [comment]
This is funny I almost wish they were always a pain to get past even if being pulled.

**Comment by coiax** [comment]
@ExcessiveUseOfCobblestone Quirks are now automatically sorted, regardless of their typepath

**Comment by ExcessiveUseOfCobblestone** [comment]
Ingame yeah but not in code. Easier to find if it's alphabetical as it is now.

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -1558,3 +1558,11 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 
 	if(channels_to_use.len)
 		world.TgsChatBroadcast()
+
+/proc/num2sign(numeric)
```
Comment (ExcessiveUseOfCobblestone) [inline code/__HELPERS/unsorted.dm]: hmm?

File: code/__HELPERS/cmp.dm [hunk 2]
```diff
@@ -81,3 +81,21 @@ GLOBAL_VAR_INIT(cmp_field, "name")
 
 /proc/cmp_advdisease_resistance_asc(datum/disease/advance/A, datum/disease/advance/B)
 	return A.totalResistance() - B.totalResistance()
+
+/proc/cmp_quirk_asc(datum/quirk/A, datum/quirk/B)
```
Comment (ExcessiveUseOfCobblestone) [inline code/__HELPERS/cmp.dm]: hmm?

Comment (coiax) [reply code/__HELPERS/cmp.dm]: Relying on type ordering will be very difficult to do if the file with the types is split or rearranged.

Comment (ExcessiveUseOfCobblestone) [reply code/__HELPERS/cmp.dm]: yep I see it now

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by PKPenguin321** [comment]
Free perk point

**Comment by ExcessiveUseOfCobblestone** [comment]
I think it's a free point up until you're in a rush and CANT PASS THIS FAT IDIOT OH MY GOODNESS WE'RE GOING TO DIE

Resting just makes you a ripe target for whatever you're running away from.

**Comment by ExcessiveUseOfCobblestone** [comment]
No fatshaming

**Pull request merged**